src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1927,6 +1927,7 @@ internal ImmutableArray<Diagnostic> GetDiagnostics(CompilationStage stage, bool
                             var syntaxTree = syntaxTrees[i];
                             AppendLoadDirectiveDiagnostics(builder, _syntaxAndDeclarations, syntaxTree);
                             builder.AddRange(syntaxTree.GetDiagnostics(cancellationToken));
+                            builder.AddRange(syntaxTree.Options.Errors);
```

Comment:
Reporting the diagnostics here unconditionally will have the unfortunate consequence of causing parse option errors to be duplicated for every tree in the compilation. Since a compilation does use the parse options (e.g. language version), even if it is an indirect input, we probably do want to report problems with the parse options if a broken parse options is used to construct a tree which is used in a compilation. However, we should only report the diagnostics for any given parse options object once.